src/display/editor/tools.js

```diff
@@ -971,6 +971,22 @@ class AnnotationEditorUIManager {
     );
   }
 
+  get highlightColors() {
+    const { _highlightColors } = this;
+    if (!_highlightColors) {
+      return shadow(this, "highlightColors", null);
+    }
+    const map = new Map();
+    const hasHCM = !!this.#pageColors;
+    for (const [name, color] of _highlightColors) {
+      if (hasHCM === name.endsWith("_HCM")) {
+        // We only want to use the colors that match the current HCM state.
+        map.set(name.replace("_HCM", ""), color);
```

Comment:
I'm probably missing something here, but the map only seems to be filled for HCM because there is no other place where the map is being edited, but it's being returned below so won't `highlightColors` always contain HCM-only colors or nothing if HCM is disabled? I'm mainly confused how this works for non-HCM now.